### DISCUSSION TITLE
Theme-based image display

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#3328] [Ideas]
Couldn't find a setup/reference page about this and the insider page also didn't show any noticeable mention about that, so I suggest this now.

The idea is a way to define images to only display while using a specific scheme.

The reason is simple: Readability. When you allow people to toggle between the schemes could it become hard to look at or recognize images, because it may use colours that conflict with the scheme currently used, so having a way to only display the image if the specific scheme is used would be a neat addition to have.

The inspiration came from GitHub implementing a way to [define a theme context for images in November 24th](https://github.blog/changelog/2021-11-24-specify-theme-context-for-images-in-markdown/) which allows you to only show specific images under specific themes.

As an example do I have here images used on the [Primer Octicons Repository](https://github.com/primer/octicons). One is only visible in dark and one only in light theme.

![octicons cover light](https://user-images.githubusercontent.com/54012/138925195-5779c51d-ff8c-4264-a914-e64f4843893d.png#gh-light-mode-only)
![octicons cover dark](https://user-images.githubusercontent.com/54012/138925203-80e1afa1-ba54-4731-9525-3c41186663f9.png#gh-dark-mode-only)

**Comment by squidfunk** [comment]
Using a hash fragment is a very interesting idea! Could you open an issue with a FR? I can look into it.

**Reply by squidfunk** [reply]
Added in e3fba7ac7. I just need to update the docs. You can now use `#only-light` and `#only-dark` as hash fragments on image URLs.

**Reply by squidfunk** [reply]
Docs added in a6d8e555c.

**Reply by Andre601** [reply]
Should the docs already show the feature?

Because I can't see it working here...
![image](https://user-images.githubusercontent.com/11576465/145681745-68c4b21e-3005-41b5-b20f-d215689bccdd.png)

**Reply by squidfunk** [reply]
Yes, they should and now they do 😅 I forgot to retrigger the build after doing the Insiders merge, which is what the documentation is built with. Sorry for the troubles, I've been on and off my notebook today.
